components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -4167,7 +4167,7 @@ private static X509Certificate getPublicCertFromJWKS(String jwksUri) throws Iden
      * @return Jwks Url
      * @throws IdentityOAuth2Exception
      */
-    private static String getSPJwksUrl(String clientId, String spTenantDomain) throws IdentityOAuth2Exception {
+    public static String getSPJwksUrl(String clientId, String spTenantDomain) throws IdentityOAuth2Exception {
```

Comment:
Check whether the ServiceProvider can directly give the JWKSUri with its API rather than going through the SP properties

Reply:
Addressed with 89289ac01f74a9042cafdfcfc57d4d4c4027f723